android/src/main/java/io/sentry/react/RNSentryBreadcrumb.java

```diff
@@ -0,0 +1,87 @@
+package io.sentry.react;
+
+import com.facebook.react.bridge.ReadableMap;
+
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Map;
+
+import io.sentry.Breadcrumb;
+import io.sentry.SentryLevel;
+
+public class RNSentryBreadcrumb {
+
+    @Nullable
+    public static String getCurrentScreenFrom(ReadableMap from) {
+        final @Nullable String maybeCategory = from.hasKey("category") ? from.getString("category") : null;
+        if (maybeCategory == null || !maybeCategory.equals("navigation")) {
+            return null;
+        }
+
+        final @Nullable ReadableMap maybeData = from.hasKey("data") ? from.getMap("data") : null;
+        if (maybeData == null) {
+            return null;
+        }
+
+        try {
+            // getString might throw if cast to string fails (data.to is not enforced by TS to be a string)
+            return maybeData.hasKey("to") ? maybeData.getString("to") : null;
```

Comment:
If the suggestions above are accepted we can change this too
```suggestion
            return data.hasKey("to") ? data.getString("to") : null;
```